### PR TITLE
TestEcdsa: fix flaky "tampering" of public key

### PR DESCRIPTION
**Pull request by cce**
## Summary

This test (TestEcdsa) tests the ecdsa_pk_decompress opcode and intentionally "tampers" with the public key by setting the first byte to zero. Occasionally this test is failing, likely because the first byte was already zero. (The test failures are for the cases where failure is expected, `pass=false`)

## Test Plan

Existing test should pass, occasional flakiness should go away.

### CODE REVIEW

**Review by jasonpaulos — DISMISSED** [review]
Nice use of overflow, looks good to me

**Comment by cce** [comment]
Thanks, that reminds me I was going to leave a comment saying "intentional overflow" — I will do that

**Review by algorandskiy — APPROVED** [review]

**Comment by algorandskiy** [comment]
Btw, there are also couple other tests that tamper input in a similar way

**Review by jasonpaulos — DISMISSED** [review]

**Review by tsachiherman — DISMISSED** [review]

**Comment by cce** [comment]
Ehh, I add a print statement to like the one below to identify whether it is pkTampered1 or pkTampered2 that was producing the most errors on master, and it is actually pkTampered2 ... so this doesn't fix that.

```
    pkTampered1 := make([]byte, len(pk))
    copy(pkTampered1, pk)
    pkTampered1[0] += 0
    pkTampered2 := make([]byte, len(pk))
    copy(pkTampered2, pk[1:])

    var decompressTests = []struct {
        key  []byte
        pass bool
    }{
        {pk, true},
        {pkTampered1, false},
        {pkTampered2, false},
    }
    for i, test := range decompressTests {
        t.Run(fmt.Sprintf("decompress/pass=%v", test.pass), func(t *testing.T) {
            t.Log("decompressTests i", i)
```

**Comment by cce** [comment]
OK, this time I read up on the 33-byte ecdsa compressed public key format and made the tampering always work.

File: data/transactions/logic/evalCrypto_test.go [hunk 1]
```diff
@@ -212,9 +212,9 @@ byte 0x%s
 &&`
 	pkTampered1 := make([]byte, len(pk))
 	copy(pkTampered1, pk)
-	pkTampered1[0] = 0
-	pkTampered2 := make([]byte, len(pk))
-	copy(pkTampered2, pk[1:])
+	pkTampered1[0] = 0                     // first byte is a prefix of either 0x02 or 0x03
+	pkTampered2 := make([]byte, len(pk)-1) // must be 33 bytes length
+	copy(pkTampered2, pk)
```
Comment (jasonpaulos) [inline data/transactions/logic/evalCrypto_test.go:217]: nit: `pkTampered2 := pk[:len(pk)-1]` would achieve the same thing

**Review by jasonpaulos — APPROVED** [review]
This makes sense to me

**Pull request merged**